dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1002 @@
+
+    Args:
+      project_id: Project Id to retrieve the list of disk images for
+
+    Return:
+      Dict of disk images
+    """
+    result: Dict[str, Resource] = {}
+
+    # Using beta version of the API because v1 did not have important
+    # information when creating this script
+    compute_api_client = CreateService('compute', 'beta')
+
+    # Disk images are not tied to a zone, so there is no aggregatedList
+    request = compute_api_client.images().list(project=project_id)  # pylint: disable=no-member
+
+    while request is not None:
+      response = request.execute()
+
+      if response:
+        for image in response.get('items', {}):
+          resource = Resource()
+          resource.id = image.get('id')
+          resource.resource_name = image.get('selfLink')
+          resource.creation_timestamp = image.get('creationTimestamp')
+          if image.get('sourceDisk'):
+            resource.parent = Resource()
+            resource.parent.resource_name = image.get('sourceDisk')
+            resource.parent.id = image.get('sourceDiskId')
+          elif image.get('sourceSnapshot'):
+            resource.parent = Resource()
+            resource.parent.resource_name = image.get('sourceSnapshot')
+            resource.parent.id = image.get('sourceSnapshotId')
+
+          result[resource.id] = resource
+
+      # pylint: disable=no-member
+      request = compute_api_client.images().list_next(
+          previous_request=request, previous_response=response)
+
+    return result
+
+  def _RetrieveListOfSnapshots(self, project_id: str) -> Dict[str, Resource]:
+    """Retrieves list of snapshots in a project.
+
+    Args:
+      project_id: Project Id to retrieve the list of snapshots for
+
+    Returns:
+      Dict of snapshots
+    """
+    result: Dict[str, Resource] = {}
+
+    # Using beta version of the API because v1 did not have important
+    # information when creating this script
+    compute_api_client = CreateService('compute', 'beta')
+
+    request = compute_api_client.snapshots().list(project=project_id)  # pylint: disable=no-member
+
+    while request is not None:
+      response = request.execute()
+
+      if response:
+        for snapshot in response.get('items', {}):
+          resource = Resource()
+          resource.id = snapshot.get('id')
+          resource.resource_name = snapshot.get('selfLink')
+          resource.creation_timestamp = snapshot.get('creationTimestamp')
+          if snapshot.get('sourceDisk'):
+            resource.parent = Resource()
+            resource.parent.resource_name = snapshot.get('sourceDisk')
+            resource.parent.id = snapshot.get('sourceDiskId')
+
+          result[resource.id] = resource
+
+      # pylint: disable=no-member
+      request = compute_api_client.snapshots().list_next(
+          previous_request=request, previous_response=response)
+
+    return result
+
+  def _RetrieveListOfInstances(self, project_id: str) -> Dict[str, Resource]:
```

Comment:
this can be replaced by function in https://github.com/google/cloud-forensics-utils/blob/main/libcloudforensics/providers/gcp/internal/compute.py

Reply:
Ya i know. At the time I wrote this part of the code some of the required fields were only in compute engine beta api and not in the v1 used by libcloudforensics. So i had to do it myself. Will try libcloudforensics and check if v1 has the fields i want.